src/database/index.tsx

```diff
@@ -406,6 +373,51 @@ export const getFacilityModelVersions = async ({
   }
 };
 
+export const getFacilities = async (
+  scenarioId: string,
+): Promise<Array<Facility> | null> => {
+  try {
+    const scenario = await getScenario(scenarioId);
+
+    if (!scenario) return null;
+
+    const db = await getDb();
+
+    const facilitiesResults = await db
+      .collection(scenariosCollectionId)
+      .doc(scenario.id)
+      .collection(facilitiesCollectionId)
+      .orderBy("name")
+      .get();
+
+    const facilities = facilitiesResults.docs.map((doc) => {
+      return buildFacility(scenario.id, doc);
+    });
+
+    await Promise.all(
```

Comment:
It might just be late and I might just be groggy but I don't follow the async flow in this little block. We are awaiting on the all call, but then also internally awaiting on each call inside of a tight-loop. Shouldn't we just need the outer await? I'm sure I'm missing something.

Reply:
this is correct

Reply:
Thanks much for the explanation, @daschi!